src/inference/src/dev/core_impl.cpp

```diff
@@ -592,6 +592,14 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             so = ov::util::load_shared_object(desc.libraryLocation.c_str());
             std::shared_ptr<ov::IPlugin> plugin_impl;
             reinterpret_cast<ov::CreatePluginFunc*>(ov::util::get_symbol(so, ov::create_plugin_function))(plugin_impl);
+            auto plugin_name = plugin_impl->get_device_name();
+
+            // Check that device plugin name is the same as requested
+            // skip for meta
+            // TODO: Create API to detect meta plugins
+            if (plugin_name != "AUTO" && plugin_name != "MULTI") {
+                OPENVINO_ASSERT(plugin_name == deviceName, "Error: Device loaded with incorrect name\"", plugin_name, deviceName);
```

Comment:
```suggestion
                OPENVINO_ASSERT(plugin_name == deviceName, deviceName, " device loaded with incorrect name: ", plugin_name);
```
?
Should give result as: `CPU device loaded with incorrect name: GPU`